### PR TITLE
optional sync and async api

### PR DESCRIPTION
**Pull request by cdetrio**
comes after https://github.com/ewasm/evm2wasm/pull/179

### CODE REVIEW

File: wasm/generateInterface.js [hunk 1]
```diff
@@ -276,14 +277,14 @@ function generateManifest (interfaceManifest) {
     (i64.store (i32.add (get_global $sp) (i32.const ${spOffset * 32 + 8 * 3})) (i64.const 0))
     (i64.store (i32.add (get_global $sp) (i32.const ${spOffset * 32 + 8 * 2})) (i64.const 0))`
 
-      if (!op.async) {
+      if (!op.async) { // TODO: double check for sync case?
         call += '(call $bswap_m128 (i32.add (i32.const 32)(get_global $sp))) drop'
```
Comment (axic) [inline wasm/generateInterface.js]: Oh, need to "fix" the `drop` here :)

File: wasm/generateInterface.js [hunk 2]
```diff
@@ -348,15 +349,23 @@ function generateManifest (interfaceManifest) {
   return json
 }
 
-const json = generateManifest(interfaceManifest)
+// generateManifest mutates the interface input, so use a copy
+const interfaceManifestCopy = JSON.parse(JSON.stringify(interfaceManifest))
+
+let syncIfaceJson = generateManifest(interfaceManifest, {'useAsyncAPI': false})
+let asyncIfaceJson = generateManifest(interfaceManifestCopy, {'useAsyncAPI': true})
```
Comment (axic) [inline wasm/generateInterface.js]: Is `generateManifest` changing the first parameter so you need a copy?

Also how about naming it `asyncInterfaceJson` or `asyncJson`?

**Comment by axic** [comment]
Apart from the minor comment this looks great.

**Comment by axic** [comment]
Rebase please.

**Comment by cdetrio** [comment]
ewasm-kernel pinned to https://github.com/ewasm/ewasm-kernel/releases/tag/async-pojo

File: index.js [hunk 3]
```diff
@@ -2,7 +2,8 @@ const BN = require('bn.js')
 const wast2wasm = require('wast2wasm')
 const ethUtil = require('ethereumjs-util')
 const opcodes = require('./opcodes.js')
-const wastFiles = require('./wasm/wast.json')
+const wastSyncIface = require('./wasm/wast.json')
+const wastAsyncIface = require('./wasm/wast-async.json')
```
Comment (axic) [inline index.js]: Same as in the generator probably better to not shorten `Interface` to `Iface`

**Pull request merged**